doc: Update migration guide with RSA config change (#13785)
nrfconnect/sdk-nrf: doc/nrf/releases_and_maturity/migration/migration_guide_2.6.rst

  * Instead of the ``CONFIG_WIFI_MGMT_SCAN_MAX_BSS_CNT`` Kconfig option, a new :kconfig:option:`CONFIG_NRF_WIFI_SCAN_MAX_BSS_CNT` Kconfig option is added.

* For samples using ``CONFIG_NRF_SECURITY``:

  * RSA keys are no longer enabled by default.

>   This reduces the code size by 30 kB for those that are not using RSA keys.
>   This will also break the configuration for those using the RSA keys without explicitly enabling an RSA key size.

Reviewer: ```suggestion
  This reduces the code size by 30 kB if not using RSA keys.
  This also breaks the configuration if using the RSA keys without explicitly enabling an RSA key size.
```
